src/main/java/com/github/osisoft/ocs_sample_library_preview/sds/StreamsClient.java

```diff
@@ -1104,6 +1105,61 @@ public String getRangeValuesInterpolated(String tenantId, String namespaceId, St
         return response;
     }
 
+    /**
+     * gets values at the indexes provided
+     * 
+     * @param tenantId    tenant to work against
+     * @param namespaceId namespace to work against
+     * @param streamId    the stream to get values from
+     * @param index       the list of indexes to retrieve values for
+     * @return string of the array of values
+     * @throws SdsError any error that occurs
+     */
+    public String getIndexCollectionValues(String tenantId, String namespaceId, String streamId,
+            ArrayList<String> index) throws SdsError {
+        URL url = null;
+        HttpURLConnection urlConnection = null;
+        String response = "";
+
+        try {
+            String indexString = "";
+            for (int i = 0; i < index.size(); i++) {
+                if (i == 0)
```

Comment:
I think this is personal preference, but usually on these types of loops I'll do the first one outside the loop, then start the incrementor at 1 instead of 0. It's known that this comparison is only going to be true once, specifically on the first one, so checking it every time until the end is a little inefficient.

Reply:
I made the change since I agree it is better outside the loop.